cranelift/entity/src/list.rs

```diff
@@ -482,6 +482,46 @@ impl<T: EntityRef + ReservedValue> EntityList<T> {
         }
     }
 
+    /// Copies a slice from an entity list in the same pool to a position in this one.
+    pub fn copy_from(
+        &mut self,
+        other: &Self,
+        slice: impl core::ops::RangeBounds<usize>,
+        index: usize,
+        pool: &mut ListPool<T>,
+    ) {
+        assert!(
+            index <= self.len(pool),
+            "attempted to copy a slice out of bounds of `self`"
+        );
+        assert_ne!(
+            self.index, other.index,
+            "cannot copy within one `EntityList`"
+        );
```

Comment:
Thank you for this assertion, it definitely makes it easier to see that the use of `copy_within` is fine. Do you plan to follow-up with a PR to remove the need for the assert?

Reply:
I'll try to get around to it, but it might be a minute